### PR TITLE
feat: Broadcast availability to same backend users only #FS-101

### PR DESCRIPTION
**Pull request by sb88k**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-101" title="FS-101" target="_blank">StoryFS-101</a>  [iOS] Disable sending status across backends
  </td></table>
  <br />
 

## What's new in this PR?

Removing recipients from other backend when broadcasting availability change

### CODE REVIEW

File: Source/Model/User/ZMUser+Availability.swift [hunk 1]
```diff
@@ -91,6 +91,9 @@ extension ZMUser {
             .prefix(remainingSlots)
 
         recipients.formUnion(teamUsers)
+        
+        guard let domain = ZMUser.selfUser(in: context).domain else { return recipients }
+        recipients = recipients.filter { $0.domain == domain }
```
Comment (typfel) [inline Source/Model/User/ZMUser+Availability.swift:96]:
```suggestion
        recipients = recipients.filter { !$0.isFederated }
```

**Review by typfel — CHANGES_REQUESTED** [review]

File: Source/Model/User/ZMUser+Availability.swift (hunk 1 above)
Comment (typfel) [reply Source/Model/User/ZMUser+Availability.swift:96]: This is shorter and handles edge cases when the domain is `nil` on the user.

**Review by typfel — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]

**Comment by sb88k** [comment]
release minor

**Comment by sb88k** [comment]
release minor

**Pull request merged**